src/linear_algebra/prod.lean

```diff
@@ -276,6 +276,24 @@ end linear_map
 
 end prod
 
+namespace linear_map
+
+variables [comm_semiring R]
+variables [add_comm_monoid M] [add_comm_monoid M₂] [add_comm_monoid M₃] [add_comm_monoid M₄]
+variables [add_comm_monoid M₅] [add_comm_monoid M₆]
+variables [module R M] [module R M₂] [module R M₃] [module R M₄]
+
+def prod_map_linear : ((M →ₗ[R] M₃) × (M₂ →ₗ[R] M₄)) →ₗ[R] ((M × M₂) →ₗ[R] (M₃ × M₄)) :=
+{ to_fun := λ f, prod_map f.1 f.2,
+  map_add' := λ f g, by {ext, repeat {simp}},
```

Comment:
Can you extract these to standalone lemmas about `prod_map`? They can go next to `prod_map_one` etc above

Reply:
I can do that for `add`, but the `smul` one requires commutativity of `R`.

Reply:
The smul one should generalize to a different ring `S`

Reply:
Can you add the lemma for zero too?

Reply:
Done, though these are actually all proved by `rfl`. Also I noticed that `prod_map_monoid_hom` could be generalized directly to a `ring_hom` (and an `alg_hom` in the commutative case).

Reply:
That generalization was going to be my next comment!